src/app/adminDashboard/spectator/page.tsx

```diff
@@ -1,12 +1,331 @@
 
-import React from 'react'
+'use client';
+import React, { useState, useEffect } from "react";
+import styles from "../../../styles/Dashboard.module.scss";
+import { sendRequest } from "@/services/auth/auth_All_Api";
+//@ts-ignore
+import { Input, Button } from 'technogetic-iron-smart-ui';
+import { Navbar } from '../../../Components/Navbar/Navbar';
+import Loader from "@/Components/Loader/Loader";
+import TableData, { SpectatorData } from '../../../Components/Table/Table';
+import { toast } from 'react-toastify';
+import Image from "next/image";
+
+
+interface FormCreate {
+  fullName: string;
+  userName: string;
+  email: string;
+  password: string;
+  role: any;
+}
+
+export default function Modal() {
+  const [spectatorData, setSpectatorData] = useState<SpectatorData[]>([]);
+  console.log("spectatorData", spectatorData)
+  const [isLoading, setIsLoading] = useState(true);
+  const [isupdateData, setIsUpdateData] = useState<any>(null)
+  const imageIcon: string = 'spectator';
+  const [getSpectatorId, setSetSpectatorId] = useState<any>()
+
+
+  const [modal, setModal] = useState(false);
+  const [formErrors, setFormErrors] = useState<FormCreate>({
+    fullName: '',
+    userName: '',
+    email: '',
+    password: '',
+    role: '64d7239c8a677a5d2e21b00d'
+  });
+
+  const getAllUsers = async () => {
+    try {
+      const tokens = localStorage.getItem('jwtToken');
```

Comment:
every api call should have a loader state and based on that a loader should be visible on that section

Reply:
done sir